modules/server/jvm/src/main/scala/kartograffel/server/domain/repository/FileRepository.scala

```diff
@@ -1,11 +1,20 @@
 package kartograffel.server.domain.repository
 
+import cats.Monad
+import cats.data.OptionT
 import fs2.Stream
-import kartograffel.shared.domain.model.File
+import kartograffel.shared.domain.model.{File, FileF, User}
 import kartograffel.shared.model.{Entity, Graffel, Id}
 
 trait FileRepository[F[_]] extends EntityRepository[F, File] {
   def attach(file: File, graffelId: Id[Graffel]): F[Entity[File]]
 
   def findByGraffel(id: Id[Graffel]): Stream[F, Entity[File]]
+
+  def findF[A](id: Id[File], f: Id[User] => F[Option[A]])(
+      implicit F: Monad[F]): F[Option[Entity[FileF[A]]]] =
+    (for {
+      file <- OptionT(findById(id))
+      a <- OptionT(f(file.value.uploadedBy))
+    } yield Entity(Id(file.id.value), file.value.copy(uploadedBy = a))).value
```

Comment:
This function shows that the type of `Id(file.id.value)` which is `Id[FileF[A]]` depends on the parameter `A` and could differ from the usual `Id[File]`. If `A` would be `Entity[User]`, the type of that id would then be `Id[FileF[Entity[User]]]`. But the type `Id[FileF[A]]` is pretty useless if it is not the same as `Id[File]`, because there are no functions that work with arbitrary `Id[FileF[A]]`.

So `Entity` in combination with model classes that contain type parameters lead to nonsensical types and requires some unsafe unwrapping and wrapping. When using something like `FileF[A]` it seems to make more sense to have an id type `FileF.Id` that is independent of `FileF`'s type parameter.